base/ref.go

```diff
@@ -80,8 +80,10 @@ func ModifyDatasetRef(ctx context.Context, r repo.Repo, current, new *repo.Datas
 		return err
 	}
 	if err := repo.CanonicalizeDatasetRef(r, current); err != nil {
-		log.Debug(err.Error())
-		return fmt.Errorf("error with existing reference: %s", err.Error())
+		if err != repo.ErrNoHistory {
```

Comment:
I'm a fan of combining the if statement:

```go
if err := repo.CanonicalizeDatasetref(r, current); err != nil && err != repo.ErrNoHistory {
  // ...
}
```

but, might be worth pushing back & keeping it as written. It calls attention to the `ErrNoHistory` check

Reply:
I'm fine with combining them, done.